### PR TITLE
Rename locust to user

### PR DESCRIPTION
**Pull request by anuj-ssharma**
#1283

### CODE REVIEW

**Comment by cyberw** [comment]
Thanks! But there are lots of places where "User" are still called "Locust", so there is more work to be done.

Can you introduce a hidden parameter -c/--client that throws an error? Thanks!

We should also discuss whether things like LocustRunner should be renamed UserRunner? I'm in favour of that.

**Comment by heyman** [comment]
> We should also discuss whether things like LocustRunner should be renamed UserRunner? I'm in favour of that.

Hmm, maybe just `Runner` ( and`LocalRunner`, `MasterRunner` and `WorkerRunner`)?

File: locust/core.py [hunk 1]
```diff
@@ -447,7 +447,7 @@ def __new__(mcs, classname, bases, class_dict):
         return type.__new__(mcs, classname, bases, class_dict)
 
 
-class Locust(object, metaclass=LocustMeta):
```
Comment (heyman) [inline locust/core.py:450]: `LocustMeta` should be renamed to `UserMeta`.

**Comment by cyberw** [comment]
@anuj-ssharma Any chance you'll have time to look at the suggested changes & fix the conflicts soon? It would be great to be able to release 1.0 soon!

**Comment by heyman** [comment]
:+1: 

All of the documentation stills needs updating as well

I currently get an exception when I run `locust --help` with this PR.

**Comment by anuj-ssharma** [comment]
Still working through this sorry !

**Comment by anuj-ssharma** [comment]
I think I've messed up with merge conflicts

**Comment by anuj-ssharma** [comment]
Still need to update the documentation

**Comment by anuj-ssharma** [comment]
@cyberw @heyman 
What are your opinions on renaming the option `locust_classes` to `user_classes` ?

**Comment by cyberw** [comment]
> @cyberw @heyman
> What are your opinions on renaming the option `locust_classes` to `user_classes` ?

Go for it!

**Comment by anuj-ssharma** [comment]
There are errors which have not been picked up by the existing tests.

**Comment by cyberw** [comment]
The last commit broke the build (seems some tests need updating), but if you fix that then I then I am +1 to merge!

**Comment by heyman** [comment]
Great work!

> The last commit broke the build (seems some tests need updating), but if you fix that then I then I am +1 to merge!

I've fixed the broken tests, but I'd like some time to go through the changes a bit more thoroughly before be merge. Hopefully I'll be able to do that later today/tonight.

**Comment by anuj-ssharma** [comment]
My last commit (which is a merge) will break tests. Will analyse and fix them.

**Comment by anuj-ssharma** [comment]
> 
> 
> My last commit (which is a merge) will break tests. Will analyse and fix them.

and fixed now !

**Comment by heyman** [comment]
Merged!

**Pull request merged**